### DISCUSSION TITLE
v1.2.0

### DISCUSSION THREAD

**Discussion by Mrs4s** [Mrs4s/go-cqhttp#2472] [General]
## 说明
这个更新包含了近期的所有更改
建议大家开始迁移计划 -> https://github.com/Mrs4s/go-cqhttp/issues/2471 

## What's Changed
* update: support sign sever up to v1.1.3 by @KomeiDiSanXian in https://github.com/Mrs4s/go-cqhttp/pull/2283
* encoding a android_id field is not essential by @zhaodice in https://github.com/Mrs4s/go-cqhttp/pull/2318
* Update download.go by @chieve-bale in https://github.com/Mrs4s/go-cqhttp/pull/2315
* Update api.go by @PSoul in https://github.com/Mrs4s/go-cqhttp/pull/2312
* fix: group not found report (#2312) by @fumiama in https://github.com/Mrs4s/go-cqhttp/pull/2337
* feat: add waitSignServer by @KomeiDiSanXian in https://github.com/Mrs4s/go-cqhttp/pull/2311
* 使用message http-timeout作为签名服务超时设置; 提供 1.1.6 版本以上 qsign 的对接支持 by @1umine in https://github.com/Mrs4s/go-cqhttp/pull/2307
* 增加签名服务器请求超时时间 by @SuperKenVery in https://github.com/Mrs4s/go-cqhttp/pull/2302
* feat: add Bearer authentication to sign server requests by @Bluefissure in https://github.com/Mrs4s/go-cqhttp/pull/2247
* 补充剩余两个未写入的登陆协议 by @HsiangNianian in https://github.com/Mrs4s/go-cqhttp/pull/2234
* Fix #2226  尝试将 replyID 识别成正确的发送者ID by @xiangxiangxiong9 in https://github.com/Mrs4s/go-cqhttp/pull/2230
* 更新docker action, 支持更多的平台 by @xiwangly2 in https://github.com/Mrs4s/go-cqhttp/pull/2217
* :bug: 修复时区不是东八区的 BUG by @li-xunhuan in https://github.com/Mrs4s/go-cqhttp/pull/2212
* 尝试使 get_msg 获取消息中的回复信息 by @xiangxiangxiong9 in https://github.com/Mrs4s/go-cqhttp/pull/2207
* fix: 修复和文档不一致的数据 by @YepYuYu in https://github.com/Mrs4s/go-cqhttp/pull/2189
* bug-report form add aPad protocol by @A-kirami in https://github.com/Mrs4s/go-cqhttp/pull/2126
* fix: #2112 add CQ code reply to db and solve recursive reply resolve by @Lu7fer in https://github.com/Mrs4s/go-cqhttp/pull/2115
* fix: 修复release 的 action 只拉上游主仓库( fork 端拉上游仓库)，同时 checkout 没有 ref 的问题 by @tom-snow in https://github.com/Mrs4s/go-cqhttp/pull/1999
* Docker: support continuous params on CMD option by @purerosefallen in https://github.com/Mrs4s/go-cqhttp/pull/1829
* 为_send_group_notice API添加公告ID的返回值 by @Yukari316 in https://github.com/Mrs4s/go-cqhttp/pull/1834
* api: rename kick message type by @PSoul in https://github.com/Mrs4s/go-cqhttp/pull/1775
* sync by @fumiama in https://github.com/Mrs4s/go-cqhttp/pull/2340
* sync: dev to master (#2340) by @fumiama in https://github.com/Mrs4s/go-cqhttp/pull/2341
* Revert #2207 by @xiangxiangxiong9 in https://github.com/Mrs4s/go-cqhttp/pull/2397
* fix #2368, which causes system lacked sufficient buffer space by @KomeiDiSanXian in https://github.com/Mrs4s/go-cqhttp/pull/2372
* Fix: SignServer TCP ping for custom port by @xiwangly2 in https://github.com/Mrs4s/go-cqhttp/pull/2353
* 修复TCP缓冲区不足问题；重构 qsign 签名服务对接部分；支持配置多个签名服务器 by @1umine in https://github.com/Mrs4s/go-cqhttp/pull/2389
* !fixup: optimize(qisgn): async operations by @YumeMichi in https://github.com/Mrs4s/go-cqhttp/pull/2415
* fix nil pointer dereference caused by nil 'cs' by @1umine in https://github.com/Mrs4s/go-cqhttp/pull/2440
* 尝试实现 #2421 by @xiangxiangxiong9 in https://github.com/Mrs4s/go-cqhttp/pull/2422
* 尝试修复首次登录时容易出现 code -10005 和 packet timed out 的问题 by @1umine in https://github.com/Mrs4s/go-cqhttp/pull/2463
* chore: sync dev to master by @fumiama in https://github.com/Mrs4s/go-cqhttp/pull/2470

## New Contributors
* @KomeiDiSanXian made their first contribution in https://github.com/Mrs4s/go-cqhttp/pull/2283
* @zhaodice made their first contribution in https://github.com/Mrs4s/go-cqhttp/pull/2318
* @chieve-bale made their first contribution in https://github.com/Mrs4s/go-cqhttp/pull/2315
* @PSoul made their first contribution in https://github.com/Mrs4s/go-cqhttp/pull/2312
* @1umine made their first contribution in https://github.com/Mrs4s/go-cqhttp/pull/2307
* @SuperKenVery made their first contribution in https://github.com/Mrs4s/go-cqhttp/pull/2302
* @HsiangNianian made their first contribution in https://github.com/Mrs4s/go-cqhttp/pull/2234
* @xiwangly2 made their first contribution in https://github.com/Mrs4s/go-cqhttp/pull/2217
* @li-xunhuan made their first contribution in https://github.com/Mrs4s/go-cqhttp/pull/2212
* @YepYuYu made their first contribution in https://github.com/Mrs4s/go-cqhttp/pull/2189
* @A-kirami made their first contribution in https://github.com/Mrs4s/go-cqhttp/pull/2126
* @Lu7fer made their first contribution in https://github.com/Mrs4s/go-cqhttp/pull/2115
* @tom-snow made their first contribution in https://github.com/Mrs4s/go-cqhttp/pull/1999

**Full Changelog**: https://github.com/Mrs4s/go-cqhttp/compare/v1.1.0...v1.2.0

<hr /><em>This discussion was created from the release <a href='https://github.com/Mrs4s/go-cqhttp/releases/tag/v1.2.0'>v1.2.0</a>.</em>

**Comment by Apple-QAQ** [comment]
感谢陪伴
